backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -365,4 +418,362 @@ public FutureExperimentRunDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
+              runValueMap.put(
+                  "environment", ModelDBUtils.getStringFromProtoObject(environmentBlob));
+              runValueMap.put("deleted", false);
+
+              String[] fieldsArr = runValueMap.keySet().toArray(new String[0]);
+              String commaFields = String.join(",", fieldsArr);
+
+              StringBuilder queryStrBuilder =
+                  new StringBuilder("insert into experiment_run ( ")
+                      .append(commaFields)
+                      .append(") values (");
+
+              for (int i = 0; i < fieldsArr.length; i++) {
+                queryStrBuilder.append(":").append(fieldsArr[i]);
+                if (i < fieldsArr.length - 1) {
+                  queryStrBuilder.append(",");
+                }
+              }
+              queryStrBuilder.append(" ) ");
+
+              LOGGER.trace("insert experiment run query string: " + queryStrBuilder.toString());
+              var query = handle.createUpdate(queryStrBuilder.toString());
+              for (Map.Entry<String, Object> objectEntry : runValueMap.entrySet()) {
+                query.bind(objectEntry.getKey(), objectEntry.getValue());
+              }
+              query.execute();
+
+              return newExperimentRun;
+            })
+        .thenCompose(
+            handle -> tagsHandler.addTags(newExperimentRun.getId(), newExperimentRun.getTagsList()),
+            executor)
+        .thenCompose(
+            handle ->
+                attributeHandler.logKeyValues(
+                    newExperimentRun.getId(), newExperimentRun.getAttributesList()),
+            executor)
+        .thenCompose(
+            handle ->
+                hyperparametersHandler.logKeyValues(
+                    newExperimentRun.getId(), newExperimentRun.getHyperparametersList()),
+            executor)
+        .thenCompose(
+            handle ->
+                metricsHandler.logKeyValues(
+                    newExperimentRun.getId(), newExperimentRun.getMetricsList()),
+            executor)
+        // TODO .thenCompose(handle -> artifactHandler.logArtifacts(newExperimentRun.getId(),
+        // newExperimentRun.getArtifactsList()), executor)
+        // TODO .thenCompose(handle -> datasetHandler.logDatasets(newExperimentRun.getId(),
+        // newExperimentRun.getDatasetsList()), executor)
+        .thenCompose(
+            handle ->
+                observationHandler.logObservations(
+                    newExperimentRun.getId(), newExperimentRun.getObservationsList(), now),
+            executor)
+        // TODO .thenCompose(handle -> featureHandler.logFeatures(newExperimentRun.getId(),
+        // newExperimentRun.getFeaturesList()), executor)
+        // TODO .thenCompose(handle -> addCodeVersionSnapShot(), executor)
+        // TODO .thenCompose(handle -> versioned_inputs, executor)
+
+        .thenCompose(unused -> InternalFuture.completedInternalFuture(newExperimentRun), executor);
```

Comment:
Don't return the argument to the caller.

Reply:
Code is modify with better solution